### ISSUE TITLE
Cache-control directives are replaced

### ISSUE THREAD

**Issue by tanguilp** [krasenyp/plug_cache_control#3]
When using PlugCacheControl several time, only the last value is actually sent (because `Plug.Conn.put_resp_headers/3` is used).

It would be nice to have an option allowing accumulating cache-control directives.

One use-case I see is having a route setting a cache-control header (for instance `s-maxage=3600`), and a subroute setting another one (for example `stale-if-error=120`) without deleting the previous one.

**Comment by krasenyp** [comment]
You've got a point. Maybe an optional boolean parameter called `merge` or something like that can be used. In order to not break existing behaviour the default value should be `false`. Something like the following.

```elixir
# At the parent route
plug PlugCacheControl, directives: [...]

# At the subroute
plug PlugCacheControl, merge: true, directives: [...]
```

The meat of the functionality is already in `PlugCacheControl.Helpers` as there's a function for merging directives into the existing header value.

**Comment by tanguilp** [comment]
The main issue is that Phoenix sets some default headers for cache control, so it still necessary to overwrite them.

I think this is doable by storing in the `Plug.Conn{}` a flag saying whether default headers were deleted or not. If so, we append to any existing cache control header, otherwise we delete the current value and replace by the new one.

**Comment by tanguilp** [comment]
> In order to not break existing behaviour the default value should be false.

I think this is the expected default behaviour (merging) and version is pre-1.0, so I guess it could be changed, given a CHANGELOG is maintained.

**Comment by krasenyp** [comment]
I finally have some time look into the library and I released a major version - 1.0.0 which implements the behaviour that we discussed.

**Comment by tanguilp** [comment]
That's awesome, thanks!
